Assignment 1/src/main/java/com/nuclei/assignment/service/itemadder/ItemAdderImpl.java

```diff
@@ -0,0 +1,205 @@
+  
+  /**
+   * The Logger.
+   */
+  private final Log logger = LogFactory.getLog(ItemAdderImpl.class);
+  
+  @Override
+  public void inputItemsFromUserInterface(final String... rawData) {
+    logger.info("New User Session");
+    try (Scanner scanner = new Scanner(System.in)) {
+      for (final String info : StringConstantsUtils.ITEM_DETAILS_INFO) {
+        System.out.println(info);
+      }
+      int itemTypeCount = 1;
+      for (final ItemType type :ItemType.values()) {
+        System.out.printf(StringConstantsUtils.ITEM_TYPES, itemTypeCount, type);
+        itemTypeCount++;
+      }
+      System.out.println(StringConstantsUtils.DIVIDER);
+      final List<ItemEntity> items = addItemsToList(scanner, rawData);
+      logger.info(String.format("Received %d items", items.size()));
+      outputItemsWithTaxToUser(items);
+      logger.info("User Session ended");
+    } catch (Exception exception) {
+      System.out.println(exception.getMessage());
+    }
+  }
+  
+  @Override
+  public void outputItemsWithTaxToUser(final List<ItemEntity> items) throws InputException {
+    System.out.println(StringConstantsUtils.DIVIDER);
+    if (items.isEmpty()) {
+      logger.error(ExceptionsConstantsUtils.NO_INPUT);
+      throw new InputException(ExceptionsConstantsUtils.NO_INPUT);
+    }
+    logger.info(String.format("Received items: \n%s", items));
+    System.out.println(StringConstantsUtils.SHOW_ITEM_LIST);
+    int itemCount = 0;
+    for (final ItemEntity item : items) {
+      itemCount += 1;
+      System.out.printf(StringConstantsUtils.SHOW_ITEM_LIST_INFO,
+          itemCount, item.getType(), item.getName(), item.getPrice(),
+          item.getQuantity(), item.getSalesTax(), item.getFinalPrice());
+    }
+  }
+  
+  private List<ItemEntity> addItemsToList(final Scanner scanner,
+                                    final String... rawData) {
+    final ArrayList<ItemEntity> items = new ArrayList<>();
+    String addMoreToItems = StringConstantsUtils.CONFIRMATION;
+    boolean rawDataAvailable = rawData.length != 0;
+    while (addMoreToItems.equals(StringConstantsUtils.CONFIRMATION)) {
+      try {
+        String[] input = rawData;
+        if (!rawDataAvailable) {
+          System.out.println(StringConstantsUtils.INSERT_ITEM_DETAILS);
+          input = scanner.nextLine().split(" ");
+        }
+        final ItemEntity item = createItem(input);
+        logger.info(String.format("Created %s", item));
+        items.add(item);
+      } catch (Exception exception) {
+        System.out.println(exception.getMessage());
+      } finally {
+        rawDataAvailable = false;
+        System.out.println(StringConstantsUtils.DIVIDER);
+        System.out.println(StringConstantsUtils.ADD_MORE_ITEMS_QUERY);
+  
+        addMoreToItems = scanner.nextLine().toLowerCase(Locale.ROOT);
+      }
+    }
+    return items;
+  }
+  
+  @SuppressWarnings("PMD.CyclomaticComplexity")
+  private ItemEntity createItem(final String... itemProperties)
```

Comment:
Don't suppress, fix the reason causing this.

Reply:
Fixed.